tool/pub.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# forward all arguments to an available version of `pub`
+
+if [[ `command -v pub` ]]; then
+  pub "$@"
+elif [[ `command -v pub.bat` ]]; then
+  pub.bat "$@"
+else
+  dart pub "$@"
```

Comment:
Is the `dart` command not supposed to replace the above in the future? E.g. should this be checked first? But I guess fine (dart docs do not seem to indicate any difference https://dart.dev/tools/dart-tool).

Reply:
It has replaced `pub`, but this check is simpler than trying to see what happens if you call `dart pub` on an old SDK (it fails)